sim-lib/src/lib.rs

```diff
@@ -337,6 +342,19 @@ pub struct Simulation {
     no_results: bool,
 }
 
+/// PayGenDisplay is a marker trait to allow creation of trait objects for printable payment generators.
+pub trait PayGenDisplay: PaymentGenerator + Display {}
+
+/// ExecutorKit contains the components required to spin up an activity configured by the user, to be used to
+/// spin up the appropriate producers and consumers for the activity.
+struct ExecutorKit {
+    source_info: NodeInfo,
+    /// We use an arc mutex here because some implementations of the trait will be very expensive to clone.
+    /// See [NetworkGraphView] for details.
+    network_generator: Arc<Mutex<dyn NetworkGenerator + Send>>,
```

Comment:
I can't get this to work for `random_activity_nodes` because we can't clone our single `NetworkGraphView` (intentionally - expensive to copy the full graph) to produce an `ExecutorKit` for each random activity.  afaik `Box<T>` is only `Clone` if `T` is, but could be missing something?

Higher level question here is: is it worthwhile to try to use this `ExecutorKit` for both random and defined activities, given:
* Defined activities: we always have one `network_generator` for each source node
* Random activities: we share one (expensive to clone) `network_generator` between all source notes

I really like being able to have a single `dispatch_producers` function for whatever type of activity we're going to run with, particularly because I'd like to have a setting where we run defined activity for some nodes and random for others (wrote up in detail in #161).

Reply:
> The one thing that struck me a bit but I couldn't really find a way around (and there may not be any tbf) is that for the pre-defined activities we are using the exact same resource twice

Yeah, definitely didn't feel natural when I wrote it. Could split up the implementations of `NetworkGenerator` (has `destination` pubkey`) and `PaymentGenrator` (has `wait` and `amount`) so that they's a little less duplication? It's not a big space saver, but seems like a reasonable logical split anyway?

Reply:
Do you mean having two structs in the same way we do for the `Random` counterpart? If so I think that will make sense.

I do think that some renaming may be worth considering though:

The `NetworkGenerator` trait is not generating any network, it's finding a destination given an optional source. Similarly, `sample_node_by_capacity` is picking a destination node, the sampling by capacity only happens if the destination is picked randomly (which makes sense, given it is where the original method comes from, but I don't think the meaning holds anymore).

`PaymentActivityGenerator` -> `RandomPaymentActivityGenerator`? I know this is longer and we all hate long class names, but the former does not reflect the actual behavior IMO.

Feel free to disregard this, but I feel it may be worth having consistent and meaningful naming at this point so we don't have to refactor down the line. Also, naming is hard, so I'm happy to brainstorm if you consider changing it 😆

Reply:
> Do you mean having two structs in the same way we do for the Random counterpart? If so I think that will make sense.

Yep! 

> I do think that some renaming may be worth considering though

Happy to do any renaming while we're here, provided I don't have to do the brainstorming :') first instinct would be: 
* `NetworkGenerator` -> `DestinationGenerator` 
* `sample_node_by_capacity` -> `choose_destination`
* `PaymentActivityGenerator` -> `RandomPaymentActivity` (long name too long for me 💀 , and then we can have a `DefinedPaymentActivity` implementation to match it)

Reply:
LGTM! 🕺

Reply:
Added all of these renames 🧹 

As always, wondering what we were thinking with the original names in the first place 😂